snapshots/overlay/overlay.go

```diff
@@ -572,3 +595,20 @@ func supportsIndex() bool {
 	}
 	return false
 }
+
+// metacopyState returns the "metacopy" option according to kernel support and containerd config.
+func metacopyState(state bool) int {
```

Comment:
I've updated the PR. I think that enum is better, especially after comment from @AkihiroSuda https://github.com/containerd/containerd/pull/7141#discussion_r917736610, due to it would be good to hide all checks in one small function that return 3-state metacopy configuration  and then use simple switch to provide right mount option. I think it is just more readable and clear.

Reply:
Thanks for review Rodrigo!